src/compose/guard.ts

```diff
@@ -1,31 +1,29 @@
-import { difference, LIBRARY_NAME, UNKNOWN_NAME } from "@shared"
+import { difference, LIBRARY_NAME, Meta$, UNKNOWN_NAME, type Name } from "@shared"
 import { Binding$ } from "@tag"
 import { Task$ } from "@task"
 import type { Resolver } from "./resolver"
 import type { Stage, Step } from "./types"
 
 type StepType = "task" | "binding"
-type NotifyContext = { type: StepType; name: string; index: number }
+type NotifyContext = { type: StepType; name: Name; index: number }
 
 const NameMap = { task: "Task", binding: "Binding" } as const
 
 const notify = {
   duplicate: ({ type, name, index }: NotifyContext) => {
-    const message = `${LIBRARY_NAME} A duplicate ${NameMap[type]} found with ID: ${name} on stage #${index + 1}.`
+    const message = `${LIBRARY_NAME} A duplicate ${NameMap[type]} found with Name: ${name} on stage #${index + 1}.`
     throw new Error(message)
   },
 
   notSatisfied: ({ type, name, index, missing: set }: NotifyContext & { missing: Set<symbol> }) => {
-    const list = Array.from(set)
-      .map((id) => id.description ?? UNKNOWN_NAME)
-      .join(", ")
+    const list = Array.from(set, (id) => id.description ?? UNKNOWN_NAME).join(", ")
 
-    const message = `${LIBRARY_NAME} Unsatisfied dependencies found for ${NameMap[type]} with ID: ${name} on stage #${index + 1}: missing ${list}.`
+    const message = `${LIBRARY_NAME} Unsatisfied dependencies found for ${NameMap[type]} with Name: ${name} on stage #${index + 1}: missing ${list}.`
     throw new Error(message)
   },
 
   unused: ({ type, name, index }: NotifyContext) => {
-    const message = `${LIBRARY_NAME} Unused ${NameMap[type]} found with ID: ${name} on stage #${index + 1}.`
+    const message = `${LIBRARY_NAME} Unused ${NameMap[type]} found with Name: ${name} on stage #${index + 1}.`
```

Comment:
```suggestion
    const message = `${LIBRARY_NAME} Unused ${NameMap[type]} found with name: ${name} on stage #${index + 1}.`
```
No reason to keep `name` capitalized